hicommonwealth/commonwealth
# Apply contest rate limits

## Link to Issue
Closes: #8466 

## Description of Changes
- Prevents thread creation if all active contests have reached the contest post limit for the user
- Returns contest data in the GET /topics route
- Applies limit in ContestWorker where user must have > 0.0005 ETH to post to contest
- Applies limit in ContestWorker where user can post up to 2 threads per contest round
- Moves a SQL query to its own query file
- Include Marcin's UI updates too

## Test Plan
### Post limit test
- Create a contest (A)
- With a wallet > 0 ETH, create 3 posts in a row
  - Ensure that on the 3rd post, the consumer shows warning "user reached post limit for contest"
- Create a new contest (B)
- With a wallet > 0 ETH, create 1 post
  - Ensure the consumer shows the limit warning for contest A, but succeeds for contest B

### ETH requirement test
- With a wallet == 0 ETH, create a post during an active contest
  - Ensure that warning is shown: "user ETH balance insufficient"

## Other Considerations
- The only condition in which thread creation is blocked upfront is when the user has reached the post limit for all active contests
